tests/jenkins/TestSignArtifacts.groovy

```diff
@@ -57,6 +64,48 @@ class TestSignArtifacts extends BuildPipelineTest {
             return helper.callClosure(closure)
         })
 
-        super.testPipeline("jenkins/sign-artifacts/sign-standalone-artifacts.jenkinsfile", "tests/jenkins/jenkinsjob-regression-files/sign-standalone-artifacts.jenkinsfile")
+        super.testPipeline("jenkins/sign-artifacts/sign-standalone-artifacts.jenkinsfile",
+                "tests/jenkins/jenkinsjob-regression-files/sign-standalone-artifacts/sign-standalone-artifacts.jenkinsfile")
+
+        verifySignArtifactsParams(helper, '.sig', 'linux', 'workspace/artifacts/')
+
+
+        def filenamesForUrls = ['dummy_1_artifact.tar.gz', 'dummy_1_artifact.tar.gz.sig',
+                                'dummy_2_artifact.tar.gz', 'dummy_2_artifact.tar.gz.sig']
+        TestPrintArtifactDownloadUrlsForStaging.verifyPrintArtifactDownloadUrlsForStagingParams(helper, filenamesForUrls,
+                'sign_artifacts_job/dummy/upload/path/20/dist/signed')
+    }
+
+    static void verifySignArtifactsParams(helper, signatureType, distributionPlatform, artifactPath) {
+        assert helper.callStack.findAll { call ->
+            call.methodName == 'signArtifacts'
+        }.size() > 0
+
+        helper.callStack.findAll { call ->
+            call.methodName == 'signArtifacts'
+        }.each { call ->
+            assertThat(call.args.signatureType.first(), notNullValue())
+            assertThat(call.args.signatureType.first(), anyOf(equalTo('.sig'), equalTo('.pgp')))
+            assertThat(call.args.distributionPlatform.first(), notNullValue())
+            assertThat(call.args.distributionPlatform.first(), anyOf(equalTo('linux')))
+            assertThat(call.args.artifactPath.first(), notNullValue())
```

Comment:
These assertions are not part of the implementation in groovy, but part of the python code.  If a new signature type is supported or one removed these assertions will be unreliable.

https://github.com/opensearch-project/opensearch-build/blob/f79e7fd2d8d39290e064fa915800d510dfee40f8/vars/signArtifacts.groovy#L9-L30

Reply:
If a new signature type is added or removed, the test should be updated accordingly, since the job would have to be updated as the python code is updated.

Reply:
Tight coupling that makes this code harder to modify.

Alternatively, to be sure jenkins workflows are correct implemented the parameter checking (e.g. only .sig and .pgp are acceptable signatures type) should be duplicated in groovy to make the relationship of acceptable values explicit.  Is this something that we should fix in the current implementation?

Reply:
I am not a 100% I understand your concern . The `signStandalone artifacts job` will not allow any other signature type to be selected. 

This lib with any other signature type will fail with the error listing accepted signature types as it tries to run `sign.sh`. Test case for job should be able to verify the accepted types for parameters. Adding another check here would only increase one more place to add accepted types in the future as more are added or removed.

As the functionality of the script improves, we can add an option to respective jenkins job, which is `signStandalone artifacts job` and the test case. The rest of the code remains the same.

Reply:
I think this code will do more harm that good in the long run and its difficult to provide direction to other test writers what level of validation they should be performing.

Reply:
Removed the checks from tests for file types